code/modules/research/nanites/nanite_programs.dm

```diff
@@ -54,6 +54,15 @@
 	//Rules that automatically manage if the program's active without requiring separate sensor programs
 	var/list/datum/nanite_rule/rules = list()
 
+	//Logic
+	//a list of logic types a nanite program's rules follow
+	var/list/static/logic = list(
+    	"AND" = NL_AND,
```

Comment:
"AND" = "AND"?

Reply:
yes. something something magic strings bad

Reply:
```suggestion
    	NL_AND,
```